python/ray/data/_internal/dataset_repr.py

```diff
@@ -0,0 +1,571 @@
+    Dict,
+    Iterable,
+    List,
+    Mapping,
+    Optional,
+    Tuple,
+    Union,
+)
+
+import numpy as np
+
+import ray
+from ray.data._internal.logical.interfaces import SourceOperator
+from ray.data.block import Block, BlockAccessor, BlockMetadata
+from ray.data.context import (
+    DEFAULT_DATASET_REPR_HEAD_COLUMNS,
+    DEFAULT_DATASET_REPR_HEAD_ROWS,
+    DEFAULT_DATASET_REPR_MAX_BYTES_LENGTH,
+    DEFAULT_DATASET_REPR_MAX_COLLECTION_ITEMS,
+    DEFAULT_DATASET_REPR_MAX_COLUMN_WIDTH,
+    DEFAULT_DATASET_REPR_MAX_COLUMNS,
+    DEFAULT_DATASET_REPR_MAX_ROWS,
+    DEFAULT_DATASET_REPR_MAX_STRING_LENGTH,
+    DEFAULT_DATASET_REPR_MAX_TENSOR_ELEMENTS,
+    DataContext,
+)
+from ray.types import ObjectRef
+
+if TYPE_CHECKING:
+    from ray.data.dataset import Dataset, Schema
+
+_DATASET_REPR_ELLIPSIS = "…"
+_DATASET_REPR_PREVIEW_MAX_DEPTH = 2
+
+__all__ = [
+    "_build_dataset_ascii_repr",
+    "_dataset_repr_config_from_context",
+]
+
+
+@dataclass
+class _DatasetReprConfig:
+    max_rows: int
+    head_rows: int
+    max_columns: int
+    head_columns: int
+    max_column_width: int
+    max_string_length: int
+    max_bytes_length: int
+    max_collection_items: int
+    max_tensor_elements: int
+
+
+def _dataset_repr_config_from_context(
+    context: Optional[DataContext],
+) -> _DatasetReprConfig:
+    def _get_value(name: str, default: int) -> int:
+        if context is None:
+            return default
+        value = getattr(context, name, default)
+        if value is None:
+            return default
+        try:
+            return int(value)
+        except (TypeError, ValueError):
+            return default
+
+    return _DatasetReprConfig(
+        max_rows=_get_value("dataset_repr_max_rows", DEFAULT_DATASET_REPR_MAX_ROWS),
+        head_rows=_get_value("dataset_repr_head_rows", DEFAULT_DATASET_REPR_HEAD_ROWS),
+        max_columns=_get_value(
+            "dataset_repr_max_columns", DEFAULT_DATASET_REPR_MAX_COLUMNS
+        ),
+        head_columns=_get_value(
+            "dataset_repr_head_columns", DEFAULT_DATASET_REPR_HEAD_COLUMNS
+        ),
+        max_column_width=_get_value(
+            "dataset_repr_max_column_width", DEFAULT_DATASET_REPR_MAX_COLUMN_WIDTH
+        ),
+        max_string_length=_get_value(
+            "dataset_repr_max_string_length", DEFAULT_DATASET_REPR_MAX_STRING_LENGTH
+        ),
+        max_bytes_length=_get_value(
+            "dataset_repr_max_bytes_length", DEFAULT_DATASET_REPR_MAX_BYTES_LENGTH
+        ),
+        max_collection_items=_get_value(
+            "dataset_repr_max_collection_items",
+            DEFAULT_DATASET_REPR_MAX_COLLECTION_ITEMS,
+        ),
+        max_tensor_elements=_get_value(
+            "dataset_repr_max_tensor_elements",
+            DEFAULT_DATASET_REPR_MAX_TENSOR_ELEMENTS,
+        ),
+    )
+
+
+@dataclass
+class _ColumnDisplaySpec:
+    name: Optional[str]
+    dtype: Optional[str]
+    is_gap: bool = False
+
+
+def _prepare_columns_for_repr(
```

Comment:
As I read the functions in this module, it wasn't obvious to me what each of the functions did without reading the implementation. Would you mind adding docstrings summarizing what each function does (not just repeating the funtion name), and describing the input and output types?